commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MeanTest.java

```diff
@@ -106,124 +180,140 @@ static Stream<Arguments> testMean() {
             Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
             Arguments.of(new double[] {Double.NEGATIVE_INFINITY, Double.MAX_VALUE},
                 Double.NEGATIVE_INFINITY),
-            Arguments.of(new double[] {Double.MAX_VALUE, 1}, Double.MAX_VALUE / 2),
-            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}, -Double.MAX_VALUE / 3),
-            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}, -Double.MAX_VALUE / 3),
-            Arguments.of(new double[] {Double.MAX_VALUE, -1}, Double.MAX_VALUE / 2),
             Arguments.of(new double[] {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY,
-                Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY}, Double.POSITIVE_INFINITY)
+                Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {-Double.MAX_VALUE, Double.POSITIVE_INFINITY},
+                Double.POSITIVE_INFINITY)
         );
     }
 
     @ParameterizedTest
-    @MethodSource(value = "testMean")
-    void testParallelStream(double[] values, double expected) {
-        double ans = Arrays.stream(values)
-                .parallel()
-                .collect(Mean::create, Mean::accept, Mean::combine)
-                .getAsDouble();
-
-        TestHelper.assertEquals(expected, ans, 5, () -> "parallel stream");
+    @MethodSource(value = "testCombine")
+    void testCombine(double[] array1, double[] array2) {
+        final double[] combinedArray = TestHelper.concatenate(array1, array2);
+        final double expected = computeExpected(combinedArray);
+        Mean mean1 = Mean.create();
+        Mean mean2 = Mean.create();
+        Arrays.stream(array1).forEach(mean1);
+        Arrays.stream(array2).forEach(mean2);
+        final double mean2BeforeCombine = mean2.getAsDouble();
+        mean1.combine(mean2);
+        TestHelper.assertEquals(expected, mean1.getAsDouble(), 5, () -> "combine");
+        Assertions.assertEquals(mean2BeforeCombine, mean2.getAsDouble());
     }
 
     @ParameterizedTest
-    @MethodSource(value = "testMean")
-    void testMeanRandomOrder(double[] values, double expected) {
+    @MethodSource(value = "testCombine")
+    void testCombineRandomOrder(double[] array1, double[] array2) {
```

Comment:
Perhaps we should shuffle across the two arrays, either instead of, or as well as what you have here. This is the shuffle across the arrays:
```java
        double[] data = TestHelper.concatenate(array1, array2);
        int n = array1.length;
        for (int i = 1; i <= 10; i++) {
            TestHelper.shuffle(rng, data);
            System.arraycopy(data, 0, array1, 0, n);
            System.arraycopy(data, n, array2, 0, array2.length);
            testCombine(array1, array2);
        }
```